lib/ansible/modules/cloud/vmware/vmware_guest.py

```diff
@@ -2034,6 +2034,12 @@ def deploy_vm(self):
                 if self.params['esxi_hostname']:
                     relospec.host = self.select_host()
                 relospec.datastore = datastore
+                
+                if self.params['convert']:
+                    if self.params['convert'] not in ['flat', 'sparse']:
+                         self.module.fail_json(msg="Parameter 'convert' accept only value flat or sparse ")
+                    else:
+                         relospec.transform = self.params['convert']
```

Comment:
I think this option is deprecated - https://www.vmware.com/support/developer/converter-sdk/conv51_apireference/vim.vm.RelocateSpec.Transformation.html